### PR TITLE
I15089 windows system properties

### LINKED ISSUE #15089: Document System Variables usage for Windows OS
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

Referring to: https://github.com/checkstyle/checkstyle/pull/15028#discussion_r1647559846

In this PR, I had modified the `severity` property of [google_checks.xml](https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml) from:
```
<property name="severity" value="warning"/>
```

to

```
<property name="severity" value="${org.checkstyle.google.severity}" default="warning"/>
```

So we should be able to dynamically change severity level of google_checks.xml using system variable in CLI by something like:

```
java -Dorg.checkstyle.google.severity=error -jar checkstyle-xx.xx.x-all.jar -c google_checks.xml Test.java
```

But this gives error. Here is an example of it:

#### Test.java
```java
public class Test {}
```

#### Output
```
$ java -Dorg.checkstyle.google.severity=error -jar .\checkstyle-10.17.0-all.jar -c .\google_checks.xml .\Test.java
Error: Could not find or load main class .checkstyle.google.severity=error
Caused by: java.lang.ClassNotFoundException: /checkstyle/google/severity=error
```

Now if I modify the `severity` property in google_checks.xml to something like:

```
<property name="severity" value="${testing}" default="warning"/>
```

This works fine. Here's an example of it:

```
$ java -Dtesting=error -jar .\checkstyle-10.17.0-all.jar -c .\google_checks.xml .\Test.java                       
Starting audit...
[ERROR] C:\checkstyle testing\TestingGoogleStyle\.\Test.java:1:1: Missing a Javadoc comment. [MissingJavadocType]
Audit done.
Checkstyle ends with 1 errors.
```

The CLI is working perfectly for this modification, we're able to change severity level from `warning` to `error`. 

We should expect the same behavior for `severity` property mentioned at the beginning:

```
<property name="severity" value="${org.checkstyle.google.severity}" default="warning"/>
```

### PR DESCRIPTION
**Pull request by romani**
Closes #15089

### CODE REVIEW

**Comment by rnveach** [comment]
See https://github.com/checkstyle/checkstyle/issues/15089#issuecomment-2183780225

**Comment by romani** [comment]
GitHub, generate website

**Comment by romani** [comment]
I know, but it costs us nothing to keep it, and it is very handy for users.
I like ready to use examples in documentation.

**Comment by romani** [comment]
@rnveach , are you ok to remove unrelated section from system property page?
Second commit

File: src/xdocs/config_system_properties.xml [hunk 1]
```diff
@@ -16,15 +16,6 @@
...
 
-    <section name="Localization Support">
-      <p>
-        Checkstyle supports localization of the output messages.
-        If your language is not supported, please consider
-        translating the messages in the <code>messages.properties</code> file. Please let us
-        know if you translate the file.
-      </p>
-    </section>
-
```
Comment (Zopsss) [inline src/xdocs/config_system_properties.xml:27]: Why we removed this? we support localization right?

File: src/xdocs/config_system_properties.xml [hunk 2]
```diff
@@ -128,6 +119,10 @@ Audit done.
 Checkstyle ends with 2 errors.
           </source>
         </p>
+        <p>
+          Attention to Windows users: system properties should be wrapped with <code>"</code>.
+          Example: <code>-D"&lt;property&gt;=&lt;value&gt;"</code>
+        </p>
```
Comment (Zopsss) [inline src/xdocs/config_system_properties.xml:125]: It is better to have this in "Notes" section

File: src/xdocs/config_system_properties.xml (hunk 1 above)
Comment (Zopsss) [inline src/xdocs/config_system_properties.xml:27]: Why you removed this? We support localization right?

File: src/xdocs/config_system_properties.xml (hunk 2 above)
Comment (Zopsss) [inline src/xdocs/config_system_properties.xml:125]: I think it is better to have this in "Notes" section

**Review by Zopsss — COMMENTED** [review]
Some questions regarding the changes:

File: src/xdocs/config_system_properties.xml (hunk 1 above)
Comment (romani) [reply src/xdocs/config_system_properties.xml:27]: it is unrelated to system properties, page is for system properties

**Comment by romani** [comment]
GitHub, generate website

File: src/xdocs/config_system_properties.xml (hunk 2 above)
Comment (romani) [reply src/xdocs/config_system_properties.xml:125]: We should make "General" section.
Briefly describe how to do this for CLI.
And have Note subsection for windows users.

**Comment by romani** [comment]
GitHub, generate website

**Comment by romani** [comment]
@rnveach  , please consider this update.
we explain how to set system variables in general

**Comment by romani** [comment]
GitHub, generate website

**Comment by romani** [comment]
@rnveach , please consider this update.

**Comment by romani** [comment]
I removed label to attract attention, as there are some agreement to extend doc, but we still need PR to be approved.

**Comment by romani** [comment]
Github, generate report for configs in PR description

**Comment by romani** [comment]
Github, generate report for RegexpHeader/all-examples-in-one

**Comment by rnveach** [comment]
Closing this. Please create a new PR with other admins as the reviewers.
https://github.com/checkstyle/checkstyle/issues/15643#issuecomment-2336777333

**Pull request merged**